CMakeLists.txt

```diff
@@ -50,6 +50,25 @@ endif()
 
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+execute_process(
+    COMMAND git describe --dirty --always 2>/dev/null
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    OUTPUT_VARIABLE OT_COMM_GIT_REVISION OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+if(OT_COMM_GIT_REVISION)
+    set(OT_COMM_VERSION "${PROJECT_VERSION}+${OT_COMM_GIT_REVISION}")
+else()
+    set(OT_COMM_VERSION "${PROJECT_VERSION}")
+endif()
+
+message(STATUS "Version: ${OT_COMM_VERSION}")
+
+add_library(commissioner-version INTERFACE)
```

Comment:
For future flags?
```suggestion
add_library(ot-comm-config INTERFACE)
```

Reply:
renamed to `commissioner-config` to follow existing targets naming.